Add gtest

pottumuusi opened the pull request:
For testing, instead of checking output printed by game
binary, run gtest tests.

Add first test which always fails.

Commit:
Add gtest

For testing, instead of checking output printed by game
binary, run gtest tests.

Add first test which always fails.

Commit:
Make failing first test pass

Closes: #11

Commit:
Reordering Makefile

1vInc commented on Makefile:
```diff
@@ -3,44 +3,68 @@ TIDY := $(shell which clang-tidy clang-tidy-8 2> /dev/null | tail -1 | xargs bas
+INCS := -I$(INCLUDE_TOP_DIR)
+TEST_INCS := -I/opt/gtest/include/
+
+# NOTE include flags __need__ to be on the left end of arguments.
```
In test flags they are not. Remove or provide explanation why they need t obe on left end of arguments.

1vInc replied:
Fix in: 472ac17

Commit:
Remove unexplained comment

Does not seem to be true. In test flags, -I is not
on the left side of arguments.

Commit:
Fix make OK print

Commit:
Remove debug print from test target steps

Commit:
Remove duplicate flags in copilation of tests

Commit:
Reorder Makefile
